src/cpp/r/RSexp.cpp

```diff
@@ -172,24 +196,58 @@ SEXP findVar(const std::string& name, const std::string& ns)
    
    return findVar(name, env);
 }
-  
+
 SEXP findFunction(const std::string& name, const std::string& ns) 
 {
+   r::sexp::Protect protect;
```

Comment:
Ah, got it. Your approach is indeed the only way out of this box then.

On Wed, Apr 1, 2015 at 11:40 AM, Kevin Ushey <notifications@github.com>
wrote:

> In src/cpp/r/RSexp.cpp
> <https://github.com/rstudio/rstudio/pull/362#discussion_r27581264>:
>
> >  SEXP findFunction(const std::string& name, const std::string& ns)
> >  {
> > +   r::sexp::Protect protect;
>
> Right -- my point is that even though the error doesn't cause a longjmp,
> it is still printed to the console, which is undesired for us here. E.g.
> some bad code:
>
> #include <Rcpp.h>using namespace Rcpp;
> void throw_error(void* data) {
>     Rf_error("Ouch!");
> }
> // [[Rcpp::export]]void test()
> {
>     R_ToplevelExec(throw_error, NULL);
>     Rcout << "Still got here!" << std::endl;
> }
> /*** Rtest()*/
>
> gives
>
> > test()
> Error: Ouch!
> Still got here!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rstudio/rstudio/pull/362/files#r27581264>.
>